### PR TITLE
gradient_acc, get_lr_scheduler, local_rank

### PR DESCRIPTION
**Pull request by yuwangrepo**
take mean loss of gradient_acc.
implement a get_lr_scheduler to support MultiStepScheduler and PolyScheduler.
deal with local_rank to support both torchrun and torch.distributed.launch.

### CODE REVIEW

**Comment by yuwangstudy** [comment]
And a consult with @nttstar .
Thanks for your amazing work!

I currenly use insightface to reproduce "ArcFace: Additive Angular Margin Loss for Deep Face Recognition".
The paper shows that the verification results of [CASIA, ResNet50, ArcFace] is (99.53, 95.56, 95.15) on (LFW, CFP-FP, AgeDB-30).
But the results of mine is only (99.23, 94.41, 94.07) on (LFW, CFP-FP, AgeDB-30).
Is the results acceptable?

I took the same settings as the paper, with the learning rate started from 0.1 and was divided by 10 at 20, 28 epoches.
The trainning was processed on four GPUs, with the total batch size of four GPUs being 512, and stopped at 32 epoches.

Looking forward to your reply.
Thanks!

**Comment by yuwangstudy** [comment]
This is my training log. Thank you!
[training.log](https://github.com/deepinsight/insightface/files/9742837/training.log)

**Comment by yuwangstudy** [comment]
BTW.
This is my training log: [training.log](https://github.com/deepinsight/insightface/files/9742837/training.log)
This is my configure file cloud.py: [cloud.txt](https://github.com/deepinsight/insightface/files/9742853/cloud.txt)
This is my bash command:
CUDA_VISIBLE_DEVICES="0,1,2,3"
torchrun --nproc_per_node=4 --nnodes=1 --node_rank=0
--master_addr="127.0.0.1" --master_port=12358
train_v2.py
configs/cloud.py

Thank you!
